felix/fv/ipip_test.go

```diff
@@ -422,31 +398,52 @@ var _ = infrastructure.DatastoreDescribe("_BPF-SAFE_ IPIP topology before adding
 						Expect(err).NotTo(HaveOccurred())
 					}
 				}
-				c.Spec.ExternalNodesCIDRList = &[]string{addr, "1.1.1.1"}
+				c.Spec.ExternalNodesCIDRList = &[]string{addr}
 				log.WithFields(log.Fields{"felixconfiguration": c, "adding Addr": addr}).Info("Updating FelixConfiguration ")
 				_, err = client.FelixConfigurations().Update(ctx, c, options.SetOptions{})
 				Expect(err).NotTo(HaveOccurred())
 			}
-			updateConfig(prevBGPSpec.IPv4Address)
+
+			updateConfig(externalClient.IP)
 
 			// Wait for the config to take
 			for _, f := range felixes {
-				if bpfEnabled {
-					Eventually(f.BPFRoutes, "5s", "200ms").Should(ContainSubstring("1.1.1.1/32"))
-				} else {
-					Eventually(func() int {
-						return getNumIPSetMembers(f.Container, "cali40all-hosts-net")
-					}, "5s", "200ms").Should(Equal(3))
+				Eventually(func() int {
+					return getNumIPSetMembers(f.Container, "cali40all-hosts-net")
+				}, "5s", "200ms").Should(Equal(3))
+			}
+
+			Eventually(func() error {
+				err := externalClient.ExecMayFail("ip", "tunnel", "add", "tunl0", "mode", "ipip")
+				if err != nil && strings.Contains(err.Error(), "SIOCADDTUNNEL: File exists") {
+					return nil
 				}
+				return err
+			}).Should(Succeed())
```

Comment:
`Succeed()` TIL you can do `Should(Succeed())` instead of `ShouldNot(HaveOccurred())`